packages/devtools/lib/src/timeline/timeline_screen.dart

```diff
@@ -276,43 +343,111 @@ class TimelineScreen extends Screen {
     // This needs to be called before we update the button states because it
     // changes the value of [offlineMode], which the button states depend on.
     framework.exitOfflineMode();
+    // Revert to the previously selected mode on offline exit.
+    _setTimelineMode(
+        frameBasedTimelineMode: timelineController.frameBasedTimelineMode);
     _updateButtonStates();
-    _destroySplitter();
   }
 
-  Future<void> _pauseRecording() async {
-    _manuallyPaused = true;
-    timelineController.pause();
+  Future<void> _pauseFrameRecording() async {
+    assert(timelineController.frameBasedTimelineMode);
+    timelineController.pause(manual: true);
     ga.select(ga.timeline, ga.pause);
     _updateButtonStates();
     await _updateListeningState();
   }
 
-  Future<void> _resumeRecording() async {
-    _manuallyPaused = false;
+  Future<void> _resumeFrameRecording() async {
+    assert(timelineController.frameBasedTimelineMode);
     timelineController.resume();
     ga.select(ga.timeline, ga.resume);
     _updateButtonStates();
     await _updateListeningState();
   }
 
+  void _startRecording() {
+    assert(!timelineController.frameBasedTimelineMode);
+    timelineController.startRecording();
+    _recordingInstructions.hidden(true);
+    _recordingStatusMessage.text = 'Recording timeline trace';
+    _recordingStatus.hidden(false);
+    _updateButtonStates();
+  }
+
+  void _stopRecording() {
+    assert(!timelineController.frameBasedTimelineMode);
+    _recordingStatusMessage.text = 'Processing timeline trace';
+    timelineController.stopRecording();
+    _recordingStatus.hidden(true);
+    _updateButtonStates();
+  }
+
+  void _setTimelineMode({@required bool frameBasedTimelineMode}) {
```

Comment:
nit: Could be nice to have an enum instead of a boolean for frameBasedTimelineMode
then it could just be 
timelineMode
and values could be frameBased, _YOUR_NAME_HERE